File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> map;

        int n = s.size();
        int m = t.size();

        if(n != m) return false;

        unordered_map<char,int> smap;
        unordered_map<char,int> tmap;
        
        for(int i = 0; i < m; i++){
            smap[s[i]]++;
            tmap[t[i]]++;
        }
        
        for(int i = 0; i < smap.size(); i++){
            if(smap[i] != tmap[i]) return false;
        }
        return true;
        
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] words = new int[26];

    for(char c : s.toCharArray()){
        words[c - 'a']++;
    }

    for(char c : t.toCharArray()){
        words[c - 'a']--;
    }

    for(int i = 0; i < 26; i++){
        if(words[i] > 0) return false;
    }

    return true;
    }
}